### PR TITLE
Pre-commit stability improvements

### PR DESCRIPTION
**Pull request by ilaner**
## Related Issues
fixes:

## Description
* In run-unit-tests command, log the case where no test fail but the container failed to run
* Disable pre-commit multiprocessing in `validate` and `format` as it starts an MDX service.
* Pass `--assume-yes` and `--no-validate` flag to `format`.
* Set `pass_filenames` to false to build related files checkout, so it will run only once

### CODE REVIEW

File: demisto_sdk/commands/pre_commit/.pre-commit-config_template.yaml [hunk 1]
```diff
@@ -52,24 +52,24 @@ repos:
...
         name: is-gitlab-changed
         description: Checks if gitlab files are changed, and checkout to current if it is.
         entry: ./Tests/scripts/is_file_up_to_date.sh .gitlab "" true
         language: script
+        pass_filenames: false
 
   - repo: https://github.com/demisto/demisto-sdk
-    # Replace with the latest release version or master
-    rev: master
+    rev: ""
```
Comment (dorschw) [inline demisto_sdk/commands/pre_commit/.pre-commit-config_template.yaml:65]: ?

File: demisto_sdk/commands/run_unit_tests/unit_tests_runner.py [hunk 2]
```diff
@@ -150,6 +151,11 @@ def unit_test_runner(file_paths: List[Path], verbose: bool = False) -> int:
                                 logger.error(
                                     f"Test for {integration_script.object_id} failed in {case.name} with error {case.result[0].message}: {case.result[0].text}"
                                 )
+                                is_test_failed = True
+                    if not is_test_failed:
```
Comment (dorschw) [inline demisto_sdk/commands/run_unit_tests/unit_tests_runner.py:155]: let's rename it to `error`,`failed` or `test_failed`

File: demisto_sdk/commands/run_unit_tests/unit_tests_runner.py [hunk 3]
```diff
@@ -150,6 +151,11 @@ def unit_test_runner(file_paths: List[Path], verbose: bool = False) -> int:
...
+                        logger.error(
+                            f"Something is wrong with {integration_script.path} with docker image {docker_image}. Container status code {status_code}, logs: {container.logs()}"
```
Comment (dorschw) [inline demisto_sdk/commands/run_unit_tests/unit_tests_runner.py:157]:
```suggestion
                            f"Error running unit tests for {integration_script.path} using {docker_image=}. Container reports  {status_code=}, logs: {container.logs()}"
```

File: demisto_sdk/commands/pre_commit/.pre-commit-config_template.yaml (hunk 1 above)
Comment (ilaner) [reply demisto_sdk/commands/pre_commit/.pre-commit-config_template.yaml:65]: This value is replaced dynamically anyway, so the "master" value here can be misleading

File: .pre-commit-hooks.yaml [hunk 4]
```diff
@@ -13,18 +13,22 @@
...
   pass_filenames: true
   language: python
+  require_serial: true
   files: *files
 
 - id: format
   name: format
   description: Format non-code files, to comply with XSOAR standards.
   entry: demisto-sdk format
+  args: ["-ud", "-y", "--no-validate"]
```
Comment (dorschw) [inline .pre-commit-hooks.yaml:23]: why `--no-validate`?

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,10 @@
 * Fixed an issue where **modeling-rules test** did not properly handle query fields that pointed to a string.
 * Fixed an issue when trying to fetch remote files when not under the content repo.
 * Fixed a validation that the **modeling-rules test** command will fail if no test data file exist.
+* Added logging in case the container fails to run in **run-unit-tests**.
```
Comment (dorschw) [inline CHANGELOG.md:10]: pull from master please

File: .pre-commit-hooks.yaml (hunk 4 above)
Comment (ilaner) [reply .pre-commit-hooks.yaml:23]: we have a validate hook (which is enabled by default), I don't want to run `validate` twice.
`demitso-sdk --format` will format the file and run the validate hooks after

File: CHANGELOG.md (hunk 5 above)
Comment (ilaner) [reply CHANGELOG.md]: pulled

**Review by dorschw — APPROVED** [review]

**Pull request merged**